src/js/actions/libraries.js

```diff
@@ -571,10 +616,16 @@ define(function (require, exports) {
         if (!currentDocument || textLayers.isEmpty()) {
             return Promise.resolve();
         }
-
+        
         var representation = element.getPrimaryRepresentation(),
-            styleData = representation.getValue("characterstyle", "data"),
-            textLayerIDs = collection.pluck(textLayers, "id"),
+            styleData = representation.getValue("characterstyle", "data");
+        
+        // To make textLayerAdapter.applyTextStyle apply text color correctly, styleData.color must be an array.
+        if (styleData.color && !(styleData.color instanceof Array)) {
```

Comment:
good catch! we were trapped by the `charStyle.adbeFont`. I added a checker to make sure we have to font data before apply the character style.